apps/studio/src/store/modules/TabModule.ts

```diff
@@ -63,22 +63,28 @@ export const TabModule: Module<State, RootState> = {
       state.lastClosedTabs.push(...tabs.map((tab) => duplicate(tab)))
     },
     setActive(state, tab?: TransportOpenTab) {
+      const tabs = state.tabs.map(t => {
+        t.active = t.id === tab.id
+        return t
+      })
       state.active = tab
+      state.tabs = tabs
     },
   },
   actions: {
     async load(context) {
       const { usedConfig } = context.rootState
       if (usedConfig?.id) {
         log.info("Loading tabs for ", context.rootState.workspaceId, usedConfig.id)
-        const tabs = await Vue.prototype.$util.send('appdb/tabs/find', {
+        let tabs = await Vue.prototype.$util.send('appdb/tabs/find', {
           options: {
             where: {
               connectionId: usedConfig.id,
               workspaceId: context.rootState.workspaceId
             }
           }
-        });
+        })
+        tabs = tabs.filter(tab => !tab.deletedAt)
```

Comment:
I know we can't really pass `deletedAt` to options from here so this is good for now. In the future, we might wanna filter deleted rows by default like using [@deleteDateColumn](https://typeorm.io/select-query-builder#querying-deleted-rows) if that doesn't break other parts of the app. Or something like `appdb/tabs/findExceptDeleted`.

Reply:
oh good catch. I didn't even know about this.